CI test randomly fails: test_utils.py::test_gas_scaling_factor

Pre-requisite to this issue: [#938](https://github.com/oceanprotocol/ocean.py/issues/938) "Streamline Config: no config.ini file, no dict, just envvars." Then, doing this issue will be much, much simpler.

Note: this test is now skipped, until the issue is fixed. Commit: https://github.com/oceanprotocol/ocean.py/commit/e60b14fba540a9db7211bf97235a2d33ee25834b

----
Example failures: 
- https://github.com/oceanprotocol/ocean.py/actions/runs/3086219479/jobs/4990349715
- https://github.com/oceanprotocol/ocean.py/actions/runs/3086219644/jobs/4990349945

This happens, despite only small changes in unrelated code (eg tweaking a README) 

Traceback:
```text
___________________________ test_gas_scaling_factor ____________________________

web3 = <web3.main.Web3 object at 0x7fe53a2d60a0>
monkeypatch = <_pytest.monkeypatch.MonkeyPatch object at 0x7fe53b5e64c0>

    @pytest.mark.unit
    def test_gas_scaling_factor(web3, monkeypatch):
        monkeypatch.setenv("GAS_SCALING_FACTOR", "5.0")
        gas_price1 = web3.eth.gas_price
        gas_price_with_scaling = get_gas_price(web3, tx=dict())
        assert gas_price_with_scaling["gasPrice"] == gas_price1 * 5
    
        web3.eth.set_gas_price_strategy(fast_gas_price_strategy)
        gas_price2 = web3.eth.generate_gas_price()
    
        monkeypatch.delenv("GAS_SCALING_FACTOR")
        polygon_web3 = get_web3("https://polygon-rpc.com")
        polygon_web3.middleware_onion.inject(geth_poa_middleware, layer=0)
        polygon_web3.eth.set_gas_price_strategy(fast_gas_price_strategy)
>       polygon_gas = polygon_web3.eth.generate_gas_price()

ocean_lib/web3_internal/test/test_utils.py:73: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/opt/hostedtoolcache/Python/3.8.13/x64/lib/python3.8/site-packages/web3/eth.py:973: in generate_gas_price
    return self._generate_gas_price(transaction_params)
/opt/hostedtoolcache/Python/3.8.13/x64/lib/python3.8/site-packages/web3/eth.py:213: in _generate_gas_price
    return self.gasPriceStrategy(self.web3, transaction_params)
/opt/hostedtoolcache/Python/3.8.13/x64/lib/python3.8/site-packages/web3/gas_strategies/time_based.py:214: in time_based_gas_price_strategy
    probabilities = _compute_probabilities(
/opt/hostedtoolcache/Python/3.8.13/x64/lib/python3.8/site-packages/eth_utils/functional.py:45: in inner
    return callback(fn(*args, **kwargs))
/opt/hostedtoolcache/Python/3.8.13/x64/lib/python3.8/site-packages/web3/gas_strategies/time_based.py:128: in _compute_probabilities
    miner_data_by_price = tuple(sorted(
/opt/hostedtoolcache/Python/3.8.13/x64/lib/python3.8/site-packages/web3/gas_strategies/time_based.py:104: in _aggregate_miner_data
    data_by_miner = groupby(0, raw_data)
cytoolz/itertoolz.pyx:116: in cytoolz.itertoolz.groupby
    ???
cytoolz/itertoolz.pyx:160: in cytoolz.itertoolz.groupby
    ???
/opt/hostedtoolcache/Python/3.8.13/x64/lib/python3.8/site-packages/web3/gas_strategies/time_based.py:95: in _get_raw_miner_data
    block = w3.eth.get_block(block['parentHash'], full_transactions=True)
/opt/hostedtoolcache/Python/3.8.13/x64/lib/python3.8/site-packages/web3/eth.py:677: in get_block
    return self._get_block(block_identifier, full_transactions)
/opt/hostedtoolcache/Python/3.8.13/x64/lib/python3.8/site-packages/web3/module.py:57: in caller
    result = w3.manager.request_blocking(method_str,
/opt/hostedtoolcache/Python/3.8.13/x64/lib/python3.8/site-packages/web3/manager.py:198: in request_blocking
    return self.formatted_response(response,
/opt/hostedtoolcache/Python/3.8.13/x64/lib/python3.8/site-packages/web3/manager.py:177: in formatted_response
    apply_null_result_formatters(null_result_formatters, response, params)
/opt/hostedtoolcache/Python/3.8.13/x64/lib/python3.8/site-packages/web3/manager.py:82: in apply_null_result_formatters
    formatted_resp = pipe(params, null_result_formatters)
cytoolz/functoolz.pyx:[66](https://github.com/oceanprotocol/ocean.py/actions/runs/3086219479/jobs/4990349715#step:7:67)6: in cytoolz.functoolz.pipe
    ???
cytoolz/functoolz.pyx:641: in cytoolz.functoolz.c_pipe
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

params = ('0xed3e9ce5a52c76ae01aa9d125a8396562912f973a4cf8f0cc468b2e922b5bb51', True)

    def raise_block_not_found(params: Tuple[BlockIdentifier, bool]) -> NoReturn:
        try:
            block_identifier = params[0]
            message = f"Block with id: {block_identifier!r} not found."
        except IndexError:
            message = "Unknown block identifier"
    
>       raise BlockNotFound(message)
E       web3.exceptions.BlockNotFound: Block with id: '0xed3e9ce5a52c76ae01aa9d125a8396562912f973a4cf8f0cc468b2e922b5bb51' not found.

/opt/hostedtoolcache/Python/3.8.13/x64/lib/python3.8/site-packages/web3/_utils/method_formatters.py:630: BlockNotFound
---------------------------- Captured stdout setup -----------------------------
sender: 0xc594c6e5def4bab63ac29eed19a134c130388f74f019bc74b8f4389df2837a58, 0xe2DD09d719Da89e5a3D0F2549c7E24566e947260, address: 0xe2DD09d719Da89e5a3D0F2549c7E24566e947260
private key: 0xc594c6e5def4bab63ac29eed19a134c130388f74f019bc74b8f4389df2837a58
public key: 0x8b1b222564fb8302e9acfa4bcd60060de03ae479fcbed38fe21e45cd39ddb3fefb60ba1b4625e00d02223027c8f485f6d77693ab62ce06289b23942c6b319ccf

sender balance: 999999.99999999994[67](https://github.com/oceanprotocol/ocean.py/actions/runs/3086219479/jobs/4990349715#step:7:68)[80](https://github.com/oceanprotocol/ocean.py/actions/runs/3086219479/jobs/4990349715#step:7:81)224
````

--- a/ocean_lib/web3_internal/test/test_utils.py
+++ b/ocean_lib/web3_internal/test/test_utils.py
@@ -57,7 +57,6 @@ def test_prepare_fixed_hash():
 
 
 @pytest.mark.unit
-@pytest.mark.skip(reason="Don't skip once fixed #942")
 def test_gas_scaling_factor(web3, monkeypatch):
     monkeypatch.setenv("GAS_SCALING_FACTOR", "5.0")
     gas_price1 = web3.eth.gas_price
@@ -69,9 +68,7 @@ def test_gas_scaling_factor(web3, monkeypatch):
 
     monkeypatch.delenv("GAS_SCALING_FACTOR")
     polygon_web3 = get_web3("https://polygon-rpc.com")
-    polygon_web3.middleware_onion.inject(geth_poa_middleware, layer=0)
-    polygon_web3.eth.set_gas_price_strategy(fast_gas_price_strategy)
-    polygon_gas = polygon_web3.eth.generate_gas_price()
+    polygon_gas = polygon_web3.eth.gas_price
     assert polygon_gas > gas_price2
 
     monkeypatch.setenv(ENV_GAS_PRICE, "30000")

